crates/pallet-dummy-precompiles-code/src/lib.rs

```diff
@@ -0,0 +1,149 @@
+    weights::Weight,
+};
+pub use pallet::*;
+use sp_core::H160;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+/// The current storage version.
+const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+
+/// The current precompiles addresses execution version.
+pub const CURRENT_EXECUTION_VERSION: u16 = 1;
+
+/// The dummy code used to be stored for precompiles addresses - 0x5F5FFD (as raw bytes).
+///
+/// This is actually a hand-crafted sequence of opcodes for a bare-bones revert.
+/// The REVERT opcode (which is FD) - it takes two arguments from the stack with PUSH0 (5F twice).
+pub const DUMMY_CODE: &[u8] = &[0x5F, 0x5F, 0xFD];
+
+// We have to temporarily allow some clippy lints. Later on we'll send patches to substrate to
+// fix them at their end.
+#[allow(clippy::missing_docs_in_private_items)]
+#[frame_support::pallet]
+pub mod pallet {
+    use frame_support::{pallet_prelude::*, sp_std::vec::Vec};
+    use frame_system::pallet_prelude::*;
+
+    use super::*;
+
+    #[pallet::pallet]
+    #[pallet::storage_version(STORAGE_VERSION)]
+    #[pallet::generate_store(pub(super) trait Store)]
+    pub struct Pallet<T>(_);
+
+    /// Configuration trait of this pallet.
+    #[pallet::config]
+    pub trait Config: frame_system::Config + pallet_evm::Config {
+        /// The list of precompiles adresses to be created at evm with dummy code.
+        type PrecompilesAddresses: Get<Vec<H160>>;
+
+        /// The current force execute ask counter.
+        type ForceExecuteAskCounter: Get<u16>;
+    }
+
+    /// The last execution version.
+    #[pallet::storage]
+    #[pallet::getter(fn last_execution_version)]
+    pub type LastExecutionVersion<T: Config> = StorageValue<_, u16, ValueQuery>;
+
+    /// The last force execute ask counter.
+    #[pallet::storage]
+    #[pallet::getter(fn last_force_execute_ask_counter)]
+    pub type LastForceExecuteAskCounter<T: Config> = StorageValue<_, u16, ValueQuery>;
+
+    #[pallet::genesis_config]
+    #[derive(Default)]
+    pub struct GenesisConfig {}
+
+    #[pallet::genesis_build]
+    impl<T: Config> GenesisBuild<T> for GenesisConfig {
+        fn build(&self) {
+            for precompile_address in &T::PrecompilesAddresses::get() {
+                pallet_evm::Pallet::<T>::create_account(*precompile_address, DUMMY_CODE.to_vec());
+            }
+
+            <LastExecutionVersion<T>>::put(CURRENT_EXECUTION_VERSION);
+            <LastForceExecuteAskCounter<T>>::put(T::ForceExecuteAskCounter::get());
+        }
+    }
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+        fn on_runtime_upgrade() -> Weight {
+            let last_execution_version = Self::last_execution_version();
+            let last_force_execute_ask_counter = Self::last_force_execute_ask_counter();
+
+            let current_force_execute_ask_counter = T::ForceExecuteAskCounter::get();
+            let mut weight = T::DbWeight::get().reads(3);
```

Comment:
```suggestion
            let mut weight = T::DbWeight::get().reads(2);
```

Only two reads from the DB.